FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/mbedtls_freertos.h

```diff
@@ -0,0 +1,193 @@
+    size_t rootCaSize;        /**< @brief Size associated with #IotNetworkCredentials.pRootCa. */
+    const char * pClientCert; /**< @brief String representing the client certificate. */
+    size_t clientCertSize;    /**< @brief Size associated with #IotNetworkCredentials.pClientCert. */
+    const char * pPrivateKey; /**< @brief String representing the client certificate's private key. */
+    size_t privateKeySize;    /**< @brief Size associated with #IotNetworkCredentials.pPrivateKey. */
+    const char * pUserName;   /**< @brief String representing the username for MQTT. */
+    size_t userNameSize;      /**< @brief Size associated with #IotNetworkCredentials.pUserName. */
+    const char * pPassword;   /**< @brief String representing the password for MQTT. */
+    size_t passwordSize;      /**< @brief Size associated with #IotNetworkCredentials.pPassword. */
+} NetworkCredentials_t;
+
+/**
+ * @brief mbedTLS Connect / Disconnect return status.
+ */
+typedef enum MbedtlsStatus
+{
+    MBEDTLS_SUCCESS = 0,         /**< Function successfully completed. */
+    MBEDTLS_INVALID_PARAMETER,   /**< At least one parameter was invalid. */
+    MBEDTLS_INSUFFICIENT_MEMORY, /**< Insufficient memory required to establish connection. */
+    MBEDTLS_INVALID_CREDENTIALS, /**< Provided credentials were invalid. */
+    MBEDTLS_HANDSHAKE_FAILED,    /**< Performing TLS handshake with server failed. */
+    MBEDTLS_API_ERROR,           /**< A call to a system API resulted in an internal error. */
+    MBEDTLS_CONNECT_FAILURE      /**< Initial connection to the server failed. */
+} MbedtlsStatus_t;
+
+/**
+ * @brief Create a TCP connection with FreeRTOS sockets.
+ *
+ * @param[out] pNetworkContext Pointer to a network context to contain the
+ * initialized socket handle.
+ * @param[in] pHostName The hostname of the remote endpoint.
+ * @param[in] port The destination port.
+ * @param[in] pNetworkCredentials Credentials for the TLS connection.
+ * @param[in] receiveTimeoutMs Receive socket timeout.
+ * @param[in] sendTimeoutMs Send socket timeout.
+ *
+ * @return #MBEDTLS_SUCCESS, #MBEDTLS_INSUFFICIENT_MEMORY, #MBEDTLS_INVALID_CREDENTIALS,
+ * #MBEDTLS_HANDSHAKE_FAILED, #MBEDTLS_API_ERROR, or #MBEDTLS_CONNECT_FAILURE.
+ */
+MbedtlsStatus_t Mbedtls_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
+                                          const char * pHostName,
+                                          uint16_t port,
+                                          const NetworkCredentials_t * pNetworkCredentials,
+                                          uint32_t receiveTimeoutMs,
+                                          uint32_t sendTimeoutMs );
+
+/**
+ * @brief Gracefully disconnect an established TCP connection.
+ *
+ * @param[in] pNetworkContext Network context containing the TCP socket handle.
+ */
+void Mbedtls_FreeRTOS_Disconnect( const NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Receives data from an established TCP connection.
```

Comment:
Can mention here that it represents the TLS version of the transport interface's `TransportRecv_t` function. 
Similarly for the `Mbedtls_FreeRTOS_send` function